test/integration/api/user-preferences.js

```diff
@@ -191,4 +191,62 @@ describe('api: user-preferences', () => {
         });
       });
   }));
+
+  describe('audit log', () => {
+    it('can log audit events for updating and deleting site property', testService(async (service, container) => {
+      const asAlice = await service.login('alice');
+
+      const { Users } = container;
+      const alice = (await Users.getByEmail('alice@getodk.org')).get();
+
+      await asAlice.put('/v1/user-preferences/site/some-preference')
+        .send({ propertyValue: 'some-value' })
+        .expect(200);
+
+      await asAlice.delete('/v1/user-preferences/site/some-preference')
+        .expect(200);
+
+      await asAlice.get('/v1/audits')
+        .expect(200)
+        .then(({ body }) => {
+          body[0].action.should.equal('user.preference.delete');
+          body[0].actorId.should.equal(5);
+          body[0].acteeId.should.equal(alice.actor.acteeId);
+          body[0].details.should.eql({ propertyName: 'some-preference', scope: 'site' });
+
+          body[1].action.should.equal('user.preference.update');
+          body[1].actorId.should.equal(5);
+          body[1].acteeId.should.equal(alice.actor.acteeId);
+          body[1].details.should.eql({ propertyName: 'some-preference', propertyValue: 'some-value', scope: 'site' });
+        });
+    }));
+
+    it('can log audit events for updating and deleting project property', testService(async (service, container) => {
+      const asAlice = await service.login('alice');
+
+      const { Users } = container;
+      const alice = (await Users.getByEmail('alice@getodk.org')).get();
+
+      await asAlice.put('/v1/user-preferences/project/1/some-preference')
+        .send({ propertyValue: 'some-value' })
+        .expect(200);
+
+      await asAlice.delete('/v1/user-preferences/project/1/some-preference')
+        .expect(200);
+
+      await asAlice.get('/v1/audits')
+        .expect(200)
+        .then(({ body }) => {
+          body[0].action.should.equal('user.preference.delete');
+          body[0].actorId.should.equal(5);
+          body[0].acteeId.should.equal(alice.actor.acteeId);
+          body[0].details.should.eql({ propertyName: 'some-preference', projectId: '1', scope: 'project' });
```

Comment:
Not a huge deal, but I'm surprised that `projectId` is string here. That's very typical for Frontend, but in Backend, I feel like ids are consistently numeric. Maybe worth a `Number.parseInt()` in the `.audit` function? If there were frames involved, then there'd probably be a more idiomatic way to do the conversion, but I'm not sure what the best way is in this case.

Reply:
This projectId is directly from a URL query parameter. We often validate that the project with a call like this, but not in this case. I could add it.

```
Projects.getById(params.projectId)
      .then(getOrNotFound)
```

Reply:
Ah yes, I feel like something like that is how we usually convert string `params` to numeric ids. It's probably also a good idea to validate that `params.projectId` corresponds to a real project. If you want to make that change, I think it could be done as part of this PR or in a follow-up PR.